.github/workflows/commitlint.yml

```diff
@@ -0,0 +1,26 @@
+name: Commitlint
```

Comment:
Unlike `rook`, in `samba-operator` we use single workflow-action yaml, which allows us to see all jobs at once (as DAG) and halt workflow as soon a a job fails. Better have this section part of `main.yaml` with proper `needs`.